src/ucp/proto/proto_am.c

```diff
@@ -35,12 +41,44 @@ static size_t ucp_proto_pack(void *dest, void *arg)
     return 0;
 }
 
-ucs_status_t ucp_proto_progress_am_bcopy_single(uct_pending_req_t *self)
+ucs_status_t
+ucp_do_am_single(uct_pending_req_t *self, uint8_t am_id,
+                 uct_pack_callback_t pack_cb, size_t max_packed_size)
+{
+    ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
+    ucp_ep_t *ep       = req->send.ep;
+    ssize_t packed_len;
+    uint64_t *buffer;
+
+    req->send.lane = ucp_ep_get_am_lane(ep);
+
+    /* if packed data can fit short active message, use it, because it should
+     * be faster than bcopy.
+     */
+    if ((max_packed_size <= ucp_ep_config(ep)->am.max_short) &&
```

Comment:
it can be simplified (if it makes sense):
`if (max_packed_size <= ucs_min(ucp_ep_config(ep)->am.max_short,
                                UCS_ALLOCA_MAX_SIZE)) {...}`

Reply:
actually i wanted to try avoiding one branch in runtime if possible

Reply:
> actually i wanted to try avoiding one branch in runtime if possible

ah, I've got it